File: c_sharp/Enumeration/Enumeration/DocumentType.cs
using PeerageHRIS.Resources;
using System.ComponentModel.DataAnnotations;

namespace PeerageHRIS.Enumeration
{
    public enum DocumentType
    {
        [Display(Name = "CV", ResourceType = typeof(Resources.CommonResources))]
        CV,
        [Display(Name = "Contract", ResourceType = typeof(Resources.CommonResources))]
        Contract,
        [Display(Name = "ID", ResourceType = typeof(Resources.CommonResources))]
        ID,
        [Display(Name = "Passport", ResourceType = typeof(Resources.CommonResources))]
        Passport,
        [Display(Name = "DrivingLicense", ResourceType = typeof(Resources.CommonResources))]
        DrivingLicense,
        [Display(Name = "AcademicCertificate", ResourceType = typeof(Resources.CommonResources))]
        AcademicCertificate,
        [Display(Name = "TrainingCertificate", ResourceType = typeof(Resources.CommonResources))]
        TrainingCertificate,
        [Display(Name = "PaySlip", ResourceType = typeof(Resources.CommonResources))]
        PaySlip,
        [Display(Name = "TaxDocument", ResourceType = typeof(Resources.CommonResources))]
        TaxDocument,
        [Display(Name = "MedicalReport", ResourceType = typeof(Resources.CommonResources))]
        MedicalReport,
        [Display(Name = "Other", ResourceType = typeof(Resources.CommonResources))]
        Other
    }
}

File: lib/core/enums/document_type.dart
// Categorizes different types of documents.

import 'package:freezed_annotation/freezed_annotation.dart';

part 'document_type.g.dart';

@JsonEnum(alwaysCreate: true)
enum DocumentType {
  @JsonValue(0)
  cv,

  @JsonValue(1)
  contract,

  @JsonValue(2)
  id, // General ID

  @JsonValue(3)
  passport,

  @JsonValue(4)
  drivingLicense,

  @JsonValue(5)
  academicCertificate,

  @JsonValue(6)
  trainingCertificate,

  @JsonValue(7)
  paySlip,

  @JsonValue(8)
  taxDocument,

  @JsonValue(9)
  medicalReport,

  @JsonValue(10)
  other,
}